core/CronArchive/QueueConsumer.php

```diff
@@ -122,6 +123,12 @@ public function __construct(LoggerInterface $logger, $websiteIdArchiveList, $cou
 
     public function getNextArchivesToProcess()
     {
+        // in case a site is deleted while archiving is running
+        if (!empty($this->idSite) && !$this->isSiteExists($this->idSite)) {
+            $this->logger->debug("Site ID = {$this->idSite} was deleted during archiving process, moving on.");
```

Comment:
@diosmosis would the invalidation still be removed?

Reply:
@tsteur No. there used to be a scheduled task that did that (removed invalidations that were over a week old), but we removed it for some reason. I can put it back or we can explicitly delete them when removing a site.

Reply:
> I can put it back or we can explicitly delete them when removing a site.
sounds good.

Sometimes people also remove sites manually by deleting an entry in the site table but can deal with this later I suppose. Then we'd need to basically on demand remove it here.